### ISSUE TITLE
UnicodeEncodeError: 'mbcs' codec can't encode characters in position 0--1: invalid character

### ISSUE THREAD

**Issue by stjohann** [wbond/package_control#1123]
Package Control suddenly stopped working in Sublime Text 3 (works fine with Sublime Text 2). Tried multiple reinstalls, shows following error about `mbcs` codec:

```
>>> import urllib.request,os,hashlib; h = '2915d1851351e5ee549c20394736b442' + '8bc59f460fa1548d1514676163dafc88'; pf = 'Package Control.sublime-package'; ipp = sublime.installed_packages_path(); urllib.request.install_opener( urllib.request.build_opener( urllib.request.ProxyHandler()) ); by = urllib.request.urlopen( 'http://packagecontrol.io/' + pf.replace(' ', '%20')).read(); dh = hashlib.sha256(by).hexdigest(); print('Error validating download (got %s instead of %s), please try manual install' % (dh, h)) if dh != h else open(os.path.join( ipp, pf), 'wb' ).write(by) 
186789
reloading plugin Package Control.1_reloader
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 78, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.3/importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1512, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 313, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 658, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
UnicodeEncodeError: 'mbcs' codec can't encode characters in position 0--1: invalid character
reloading plugin Package Control.2_bootstrap
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 78, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.3/importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1512, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 313, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 658, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
UnicodeEncodeError: 'mbcs' codec can't encode characters in position 0--1: invalid character
reloading plugin Package Control.Package Control
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 78, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.3/importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1512, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 313, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 658, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
UnicodeEncodeError: 'mbcs' codec can't encode characters in position 0--1: invalid character
```

**Comment by FichteFoll** [comment]
What is the output of `sublime.packages_path()` in the console?

**Comment by stjohann** [comment]
`'C:\\Users\\Олег\\AppData\\Roaming\\Sublime Text 3\\Packages'` (Cyrillic characters included, yeah). Never been a problem before.

**Comment by FichteFoll** [comment]
Looks like a bug in Python http://stackoverflow.com/a/8804324/1327727.

**Comment by FichteFoll** [comment]
Python 3.3 (within ST):
```python
>>> compile('pass', r'C:\Олег\test.py', 'exec')
Traceback (most recent call last):
  File "<string>", line 1, in <module>
UnicodeEncodeError: 'mbcs' codec can't encode characters in position 0--1: invalid character
```

Python 3.5:
```python
>>> compile('pass', r'C:\Олег\test.py', 'exec')
<code object <module> at 0x000000000356AD20, file "C:\Олег\test.py", line 1>
```

**Comment by stjohann** [comment]
Well, I guess so. I suppose I’ll just have to find older version of Sublime Text 3 then (I suppose previous version didn’t have such problems with it, because even Python 2.6 on Sublime Text 2 didn’t), thanks.

**Comment by wbond** [comment]
All versions of ST3 use Python 3.3.

I looked at the bug report, but it said it applied to 3.2 and was fixed, so it should be part of 3.3.

I'll see if I can dig a little more and figure out what is going on with mbcs. Alternatively we may be able to special case the encoding ourself.

**Comment by wbond** [comment]
This is something that has to be patched in Sublime Text core. It should be fixed in the next dev release.

**Comment by FichteFoll** [comment]
My code from earlier works in 3121:

```py
>>> compile('pass', r'C:\Олег\test.py', 'exec')
<code object <module> at 0x0000000006655810, file "C:\Олег\test.py", line 1>
```

**Comment by wbond** [comment]
Yes, we pulled in some 3.4 patches to fix this in 3.3.
